addons/mail/static/src/models/autocomplete_input_view.js

```diff
@@ -17,7 +38,54 @@ registerModel({
                 return;
             }
         },
-         /**
+        /**
+         * @private
+         * @param {MouseEvent} ev
+         */
+        onKeydown(ev) {
+            if (ev.key === 'Escape') {
+                this.onBlur();
+            }
+        },
+        /**
+         * @private
+         * @param {MouseEvent} ev
+         */
+        onSelect(ev, ui) {
+            if (this.chatWindowOwnerAsNewMessage) {
+                this.chatWindowOwnerAsNewMessage.onAutocompleteSelect(ev, ui);
+                return;
+            }
+            if (this.discussSidebarCategoryOwnerAsAddingItem) {
+                this.discussSidebarCategoryOwnerAsAddingItem.onAddItemAutocompleteSelect(ev, ui);
+                return;
+            }
+            if (this.discussViewOwnerAsMobileAddItemHeader) {
+                this.discussViewOwnerAsMobileAddItemHeader.onMobileAddItemHeaderInputSelect(ev, ui);
+                return;
+            }
+            if (this.messagingMenuOwnerAsMobileNewMessageInput) {
+                this.messagingMenuOwnerAsMobileNewMessageInput.onMobileNewMessageInputSelect(ev, ui);
+                return;
+            }
+        },
+        /**
+         * @private
+         * @param {Object} req
+         * @param {function} res
+         */
+        onSource(req, res) {
+            if (this.chatWindowOwnerAsNewMessage) {
+                this.chatWindowOwnerAsNewMessage.onAutocompleteSource(req, res);
+            } else if (this.discussSidebarCategoryOwnerAsAddingItem) {
+                this.discussSidebarCategoryOwnerAsAddingItem.onAddItemAutocompleteSource(req, res);
+            } else if (this.discussViewOwnerAsMobileAddItemHeader) {
+                this.discussViewOwnerAsMobileAddItemHeader.onMobileAddItemHeaderInputSource(req, res);
+            } else if (this.messagingMenuOwnerAsMobileNewMessageInput) {
+                this.messagingMenuOwnerAsMobileNewMessageInput.onMobileNewMessageInputSource(req, res);
+            }
```

Comment:
Now that you moved the common part in a patch, you can drop the else-if forest and use the if-return pattern instead as done in the other methods.